sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -349,8 +349,18 @@ class Analyzer(override val catalogManager: CatalogManager)
       case p: LogicalPlan => p.transformExpressionsUpWithPruning(
         _.containsPattern(BINARY_ARITHMETIC), ruleId) {
         case a @ Add(l, r, f) if a.childrenResolved => (l.dataType, r.dataType) match {
-          case (DateType, _: DayTimeIntervalType) => TimeAdd(Cast(l, TimestampType), r)
-          case (_: DayTimeIntervalType, DateType) => TimeAdd(Cast(r, TimestampType), l)
+          case (DateType, dit: DayTimeIntervalType) =>
```

Comment:
The change in `Analyzer.scala` doesn't seem related to the issue raised in SPARK-35727.
If we define the arithmetic operations between `DateType` and `DayTimeIntervalType`, we should define such operators for `YearMonthIntervalType` right?

Reply:
I agree with change but I think it's a separate thing and can be done in a new PR.

The impl detail can also be improved: we can convert the interval to the number of days and then use `DateAdd` expression.

Reply:
> I agree with change but I think it's a separate thing and can be done in a new PR.
> 
> The impl detail can also be improved: we can convert the interval to the number of days (`ExtractANSIIntervalDays`) and then use `DateAdd` expression.

@cloud-fan Could I propose a ticket for improved for `DateType +/- DayTimeIntervalType(DAY)` and try to resolve it ?

Reply:
Yes please